ic-agent/src/agent/mod.rs

```diff
@@ -1554,6 +1559,43 @@ impl<'agent> QueryBuilder<'agent> {
                 self.arg,
                 self.ingress_expiry_datetime,
                 self.use_nonce,
+                None,
+            )
+            .await
+    }
+
+    /// Make a query call with signature verification. This will return a byte vector.
+    ///
+    /// Compared with [call][Self::call], this method will **always** verify the signature of the query response
+    /// regardless the Agent level configuration from [AgentBuilder::with_verify_query_signatures].
+    pub async fn call_with_verification(self) -> Result<Vec<u8>, AgentError> {
+        self.agent
+            .query_raw(
+                self.canister_id,
+                self.effective_canister_id,
+                self.method_name,
+                self.arg,
+                self.ingress_expiry_datetime,
+                self.use_nonce,
+                Some(true),
+            )
+            .await
+    }
+
+    /// Make a query call without signature verification. This will return a byte vector.
+    ///
+    /// Compared with [call][Self::call], this method will **never** verify the signature of the query response
+    /// regardless the Agent level configuration from [AgentBuilder::with_verify_query_signatures].
+    pub async fn call_without_verification(self) -> Result<Vec<u8>, AgentError> {
```

Comment:
Why are these additional build methods instead of an additional field to the builder?

Reply:
The query is conceptually the same. The difference is how do we process the response.

For similarity, `UpdateBuilder` has `call()` and `call_and_wait()`.